src/drivers/rpm/pcf8583/PCF8583.cpp

```diff
@@ -51,39 +51,72 @@ int PCF8583::init()
 		  _param_pcf8583_reset.get(),
 		  _param_pcf8583_magnet.get());
 
-	// set counter mode
-	setRegister(0x00, 0b00100000);
-
-	// start measurement
-	resetCounter();
-
-	_rpm_pub.advertise();
-
+	initCounter();
 	ScheduleOnInterval(_param_pcf8583_pool.get());
+	_rpm_pub.advertise();
 
 	return PX4_OK;
 }
 
 int PCF8583::probe()
 {
+	setRegister(0x00, 0b00100000);
+
+	uint8_t s = readRegister(0x00);
+	PX4_INFO("status register: %" PRId8 " fail_count: %" PRId8, s, _tranfer_fail_count);
+
+	setRegister(0x04, 10);
```

Comment:
Maybe worth adding a quick comment what you're doing here? Otherwise it's a lot of magic numbers.

Reply:
I've improved comments throughout code